aws-api/src/test/java/com/amplifyframework/api/aws/AWSApiPluginTest.java

```diff
@@ -246,6 +255,13 @@ public void graphQlMutationGetsResponse() throws JSONException, ApiException {
 
         // Assert that the expected response was received
         assertEquals(expectedName, actualResponse.getData().getName());
+
+        // Verify that the expected hub event fired.
+        List<HubEvent<?>> events = networkStatusObserver.await();
+        assertTrue(events.size() > 0);
+        assertTrue(events.get(0).getData() instanceof ApiEndpointStatusChangeEvent);
+        ApiEndpointStatusChangeEvent eventData = (ApiEndpointStatusChangeEvent) events.get(0).getData();
```

Comment:
**Aha!**

We have this unfortunate situation with Hub right now, where we have to cast the data type.

I wonder if instead we can include a factory on the event structure, like:
```java
ApiEndpointStatusChangedEvent event = ApiEndpointStatusChangedEvent.fromEvent(events.get(0));
```
Or, very similar:
```java
ApiEndpointStatuChangedEvent event = ApiEndpointStatusChangedEvent.from(events.get(0).getData());
```

This would at least pull the cast into our own code where we can do it safely, without the customer needing to worry about it.

Impl might start off looking like an `equals(....)` almost:
```
public static ApiEndpointStatusChangedEvent from(Object someBlob) {
    if (!(someBlob instanceof ApiEndpointStatuChangedEvent)) {
        throw BadNewsBearsError("Shucks!");
    }
    ... etc. validate fields,
    return new ApiEndpointStatusChangeEvent(..., ...);
}
```

Reply:
I created a couple of factory methods in each of the classes used as payloads for hub events. There's definitely some work to do in the Hub in general to make the approach cleaner. As it stands, the `subscribe` method definition takes a `HubSubscriber` instance which is currently just taking `HubEvent<?>` as a parameter which kind of prevents a cleaner approach to the problem. 

TLDR: a cleaner approach would probably require changes to the Hub category definition.

Reply:
I did some refactoring for this to make the calling code a bit cleaner by adding a factory method to `HubSubscriber`. Calling code then looks like this: 

```java
HubSubscriber subscriber = HubSubscriber.<ApiEndpointStatusChangeEvent>create(eventData -> {
	NetworkStatusEvent.from(eventData).toHubEvent().publish(HubChannel.DATASTORE, Amplify.Hub);
};

HubFilter filter = hubEvent -> {
	return ApiChannelEventName.API_ENDPOINT_STATUS_CHANGED.name().equals(hubEvent.getName());
};

Amplify.Hub.subscribe(HubChannel.API, filter, subscriber);
```